apps/modernization-ui/src/apps/page-builder/page/management/edit/question/GroupQuestion/GroupQuestion.tsx

```diff
@@ -0,0 +1,150 @@
+    const init = {
+        name: subsection.name,
+        batches: questions.map((question) => ({
+            batchTableAppearIndCd: undefined,
+            batchTableColumnWidth: undefined,
+            batchTableHeader: undefined,
+            id: question.id
+        })),
+        blockName: undefined,
+        id: subsection.id,
+        visibleText: subsection.visible ? 'Y' : 'N',
+        repeatNumber: 1
+    };
+    const methods = useForm<GroupQuestionFormType>({
+        mode: 'onChange',
+        defaultValues: { ...init }
+    });
+    const { handleSubmit, reset } = methods;
+
+    const handleGroup = (data: GroupQuestionFormType) => {
+        const request = {
+            batches: data.batches,
+            blockName: data.blockName,
+            id: data.id
+        };
+        try {
+            SubSectionControllerService.groupSubSectionUsingPost({
+                authorization: authorization(),
+                page: page.id,
+                request: request
+            });
+        } catch (error) {
+            if (error instanceof Error) {
+                console.error(error);
+                setAlert({
+                    type: 'error',
+                    expiration: 5000,
+                    message: <p>{error.message}</p>
+                });
+            } else {
+                console.error(error);
+                setAlert({
+                    type: 'error',
+                    expiration: 5000,
+                    message: <p>An unknown error occurred</p>
+                });
+            }
+        } finally {
+            setAlert({
+                type: 'success',
+                expiration: 5000,
+                message: (
+                    <p>
+                        You've successfully grouped <span>{data.blockName}</span>
+                    </p>
+                )
+            });
+            refresh();
+        }
+    };
+
+    const handleSubsectionUpdate = (data: GroupQuestionFormType) => {
+        const request = {
+            name: data.name,
+            visible: data.visibleText === 'Y' ? true : false,
+            isGrouped: true
+        };
+        try {
+            SubSectionControllerService.updateSubSectionUsingPut({
+                authorization: authorization(),
+                page: page.id,
+                request: request,
+                subSectionId: subsection.id
+            });
+        } catch (error: unknown) {
+            if (error instanceof Error) {
+                console.error(error);
+                setAlert({
+                    type: 'error',
+                    expiration: 5000,
+                    message: <p>{error.message}</p>
+                });
+            } else {
+                console.error(error);
+                setAlert({
+                    type: 'error',
+                    expiration: 5000,
+                    message: <p>An unknown error occurred</p>
+                });
+            }
+        }
+    };
```

Comment:
Question: Seems very odd to allow "editing" the subsection at the same time as grouping it. I get that its the way Classic handles it, just don't think I'm a fan.

Reply:
I know, it's weird and annoying. Don't know what to say about that :)